tests/integration_tests/build/test_kani.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""
+Proofs ensuring memory safety properites, user-defined assertions,
+absence of panics and some types of unexpected behavior (e.g., arithmetic overflows).
+"""
+import concurrent.futures
+import platform
+
+import pytest
+
+from framework import utils
+
+PLATFORM = platform.machine()
+CRATES_WITH_PROOFS = ["dumbo", "rate_limiter"]
+
+
+@pytest.mark.timeout(1800)
+@pytest.mark.skipif(PLATFORM != "x86_64", reason="Kani proofs run only on x86_64.")
+def test_kani():
```

Comment:
Another approach would be to parametrize this test with the list of proofs, and then in `pipeline_pr.py` use xdist to run the tests in parallel. Just append `-nauto`